internal/config/node/node.go

```diff
@@ -0,0 +1,58 @@
+// +build linux
+
+package node
+
+import (
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// ValidateConfig initializes and validates all of the singleton variables
+// that store the node's configuration.
+// Currently, we check hugetlb, cgroup v1 or v2, pid and memory swap support for cgroups.
+// We check the error at server configuration validation, and if we error, shutdown
+// cri-o early, instead of when we're already trying to run containers.
+func ValidateConfig() error {
+	toInit := []struct {
+		name        string
+		initializer func() bool
+		err         *error
+		fatal       bool
+	}{
+		{
+			name:        "hugetlb cgroup",
+			initializer: CgroupHasHugetlb,
+			err:         &cgroupControllerErr,
+			fatal:       false,
```

Comment:
nit: you don't have to initialize struct fields which have default value, and for `bool` the default is `false`. It some cases omitting the default might hurt readability but I doubt it's the case here.

Reply:
there are no fields that are fatal, so I've dropped that for now